### PR TITLE
Fixes #26979 - Remove Ruby 1.x compatibility code

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: lib/proxy/util.rb [hunk 1]
```diff
@@ -71,24 +71,7 @@ def which(bin, *path)
   end
 
   def escape_for_shell(command)
```
Comment (ekohl) [inline lib/proxy/util.rb:73]: I considered removing this, but it's mostly used in the Puppetrun code so I'd prefer to revisit this after https://github.com/theforeman/smart-proxy/pull/637

**Review by tbrisker — APPROVED** [review]
Thanks @ekohl !

**Comment by tbrisker** [comment]
[test core]

**Comment by tbrisker** [comment]
Test failures related: 
`File does not exist: checks`
guessing some places require the files deleted here?

**Comment by ekohl** [comment]
Was green, but needed a rebase

**Comment by tbrisker** [comment]
Thanks @ekohl ! about time we got rid of those :)

**Comment by lzap** [comment]
Holy :shield: it's happening!

**Comment by lzap** [comment]
Guys, after this patch smart proxy won't start:

```
[lzap@box smart-proxy]$ be bin/smart-proxy
bundler: failed to load command: bin/smart-proxy (bin/smart-proxy)
LoadError: cannot load such file -- daemon
  /home/lzap/work/smart-proxy/lib/smart_proxy_main.rb:9:in `require'
  /home/lzap/work/smart-proxy/lib/smart_proxy_main.rb:9:in `<top (required)>'
  bin/smart-proxy:5:in `require'
  bin/smart-proxy:5:in `<top (required)>'
```

Taking a look.

**Comment by ekohl** [comment]
Oh yes, I removed the `lib/daemon.rb` workaround file but the require is still there. Also working on it :)

**Comment by lzap** [comment]
Merged thanks!

**Pull request merged**